token/token.go

```diff
@@ -0,0 +1,158 @@
+package token
+
+import (
+	"crypto/ecdsa"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	b64 "encoding/base64"
+
+	"github.com/NBISweden/sda-uppmax-integration/helpers"
+	"github.com/golang-jwt/jwt"
+	log "github.com/sirupsen/logrus"
+)
+
+type tokenRequest struct {
+	Swamid    string `json:"swamid"`
+	ProjectID string `json:"projectid"`
+}
+
+type tokenResponse struct {
+	Swamid      string `json:"swamid"`
+	ProjectID   string `json:"projectid"`
+	RequestTime string `json:"request_time"`
+	Expiration  string `json:"expiration"`
+	S3Config    string `json:"s3config"`
+}
+
+func readRequestBody(body io.ReadCloser) (tokenRequest tokenRequest, err error) {
+
+	reqBody, err := ioutil.ReadAll(body)
+	if err != nil {
+		return tokenRequest, fmt.Errorf("Reading request body")
+	}
+	defer body.Close()
+
+	err = json.Unmarshal(reqBody, &tokenRequest)
+	if err != nil {
+		return tokenRequest, fmt.Errorf("Unmarshaling data")
+	}
+
+	if tokenRequest.ProjectID == "" || tokenRequest.Swamid == "" {
+		return tokenRequest, fmt.Errorf("Incomplete data")
+	}
+
+	return tokenRequest, nil
+}
+
+func createECToken(key *ecdsa.PrivateKey, username string) (string, error) {
+	// signing method of token
+	token := jwt.New(jwt.SigningMethodES256)
+	// token headers
+	token.Header["alg"] = "ES256"
+	// token claims
+	claims := make(jwt.MapClaims)
+	claims["iss"] = helpers.Config.Iss
+
+	claims["exp"] = time.Now().AddDate(0, 0, helpers.Config.ExpirationDays).Unix()
+	claims["sub"] = username
+	token.Claims = claims
+
+	// create token
+	tokenString, err := token.SignedString(key)
+	if err != nil {
+		return "no-token", err
+	}
+
+	return tokenString, nil
+}
+
+func createS3Config(username string) (s3config string, expiration string, err error) {
+	s3config = "guess_mime_type = True \nhuman_readable_sizes = True\nuse_https = True\n" +
+		"multipart_chunk_size_mb = 50\n" + "check_ssl_certificate = False\n" +
+		"check_ssl_hostname = False\n" + "encoding = UTF-8\n" + "encrypt = False\n" +
+		"socket_timeout = 30\n"
```

Comment:
```suggestion
	s3config = "guess_mime_type = True \nhuman_readable_sizes = True\nuse_https = True\n" +
		"multipart_chunk_size_mb = 50\n" + "check_ssl_certificate = True\n" +
		"check_ssl_hostname = True\n" + "encoding = UTF-8\n" + "encrypt = False\n" +
		"socket_timeout = 30\n"
```
certificate validation should be performed when TLS is enabled

Reply:
The values are the ones we are using when creating the s3config, at least in `login.test.ega.nbis.se`. If that's wrong, I guess we need to change it there too

Reply:
Yes, always validate the certificates.